bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1348,7 +1350,7 @@ public ImageData getImageData() {
 public ImageData getImageData (int zoom) {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
 
-	if (zoom == currentDeviceZoom) {
+	if (zoom == getZoom()) {
```

Comment:
Similar to https://github.com/eclipse-platform/eclipse.platform.swt/pull/1229#discussion_r1601379823, please extract all calls to `getZoom()` inside this method to a variable called `oldZoom` to improve readability. It makes sense in this case since the method receives a (new/targeted) `zoom` as parameter and the method `getZoom()` retrieves the current/soon-to-be-old zoom.

Reply:
already addressed in the comment above :)